### ISSUE TITLE
Doing what the help messages indicates results in an error

### ISSUE THREAD

**Issue by blyxyas** [rust-lang/rust-clippy#10671]
### Summary

Doing the following command will result in an error:

```sh
cargo clippy -- -W clippy::pedantic
```

Even though using that command is what Clippy's help message (`cargo clippy -h`) indicates you to do?

Note that I used `clippy::pedantic` but any lint or lint group will result in an error.

This is either a documentation error (the help messages indicates how to get an error) or a normal bug

<details><summary>Output</summary>

```
error: failed to run `rustc` to learn about target-specific information

Caused by:
  process didn't exit successfully: `/home/alejandra/.rustup/toolchains/nightly-2023-04-06-x86_64-unknown-linux-gnu/bin/clippy-driver rustc - --crate-name ___ --print=file-names -Zunstable-options -Zbinary-dep-depinfo --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro --print=sysroot --print=split-debuginfo --print=crate-name --print=cfg` (exit status: 1)
  --- stdout

  The flag `-Wall` does not exist in `rustc`. Most useful lints are enabled by
  default. Use `rustc -W help` to see all available lints. It's more common to put
  warning settings in the crate root using `#![warn(LINT_NAME)]` instead of using
  the command line flag directly.

```

</details>

### Reproducer

**N/A**

### Version

```text
rustc 1.70.0-nightly (2eaeb1eee 2023-04-05)
binary: rustc
commit-hash: 2eaeb1eee1b21772de8b935236d16ff8e03fdcf5
commit-date: 2023-04-05
host: x86_64-unknown-linux-gnu
release: 1.70.0-nightly
LLVM version: 16.0.0
```

### Additional Labels

@rustbot label +C-bug

**Comment by Alexendoo** [comment]
Do you get the same error if you run the failing command directly? (with `< /dev/null`)

**Comment by blyxyas** [comment]
Gotta completely update the issue. It seems like now every use of `cargo clippy` is corrupt?
I'm going to test this further and come up with a MRE. Then I'll update the issue.

**Comment by blyxyas** [comment]
I'm going to close this.
After running `cargo check`, stopping it mid-check and **then** running `cargo clippy`. It worked again.

Not a single bit of idea about what could even start to explain what happened here. Probably dark magic.
